File: ios/Classes/Utility.m
#import "Utility.h"

@implementation Utility

+ (NSString *)basePath {
    NSFileManager *fileManager = [NSFileManager defaultManager];
    
    NSString *tempPath = NSTemporaryDirectory();
    NSString *path = [tempPath stringByAppendingString:@"video_compress"];
    
    if (![fileManager fileExistsAtPath:path]){
        [fileManager createDirectoryAtPath:path withIntermediateDirectories:true attributes:nil error:nil];
    }
    return path;
}

+ (NSString *)excludeFileProtocol:(NSString *)path {
    return [path stringByReplacingOccurrencesOfString:@"file://" withString:@""];
}

+ (NSURL *)getPathUrl:(NSString *)path {
    return [NSURL fileURLWithPath:[self excludeFileProtocol:path]];
}

+ (NSString *)stripFileExtension:(NSString *)fileName {
    return [fileName stringByDeletingPathExtension];
}

+ (NSString *)getFileName:(NSString *)path {
    return [self stripFileExtension:[path lastPathComponent]];
}

+ (void)deleteFile:(NSString *)path {
    NSFileManager *fileManager = [NSFileManager defaultManager];
    NSURL *url = [self getPathUrl:path];
    [fileManager removeItemAtURL:url error:nil];
}

+ (NSString *)keyValueToJson:(NSDictionary *)keyAndValue {
    NSData *data = [NSJSONSerialization dataWithJSONObject:keyAndValue options:NSJSONWritingFragmentsAllowed error:nil];
    return [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
}

@end

File: ios/Classes/Utility.swift

class Utility: NSObject {
    static let fileManager = FileManager.default
    
    static func basePath()->String {
        let path = "\(NSTemporaryDirectory())video_compress"
        do {
            if !fileManager.fileExists(atPath: path) {
                try! fileManager.createDirectory(atPath: path,
                                                 withIntermediateDirectories: true, attributes: nil)
            }
        }
        return path
    }
    
    static func stripFileExtension(_ fileName:String)->String {
        var components = fileName.components(separatedBy: ".")
        if components.count > 1 {
            components.removeLast()
            return components.joined(separator: ".")
        } else {
            return fileName
        }
    }
    static func getFileName(_ path: String)->String {
        return stripFileExtension((path as NSString).lastPathComponent)
    }
    
    static func getPathUrl(_ path: String)->URL {
        return URL(fileURLWithPath: excludeFileProtocol(path))
    }
    
    static func excludeFileProtocol(_ path: String)->String {
        return path.replacingOccurrences(of: "file://", with: "")
    }
    
    static func excludeEncoding(_ path: String) -> String{
        return path.removingPercentEncoding!
    }
    
    static func keyValueToJson(_ keyAndValue: [String : Any?])->String {
        let data = try! JSONSerialization.data(withJSONObject: keyAndValue as NSDictionary, options: [])
        let jsonString = NSString(data:data as Data,encoding: String.Encoding.utf8.rawValue)
        return jsonString! as String
    }
    
    static func deleteFile(_ path: String, clear: Bool = false) {
        let url = getPathUrl(path)
        if fileManager.fileExists(atPath: url.absoluteString) {
            try? fileManager.removeItem(at: url)
        }
        if clear {
            try? fileManager.removeItem(at: url)
        }
    }
}
